lib/utils/web_link_generator.dart

```diff
@@ -0,0 +1,198 @@
+import 'dart:developer';
+
+/// Utility class for generating web links in flat subdomain format.
+///
+/// This class generates HTTPS URLs by combining a workplace FQDN with an
+/// optional app slug in a flat subdomain structure (e.g., `alice-notes.example.com`).
+class WebLinkGenerator {
+  /// Maximum length for FQDN to prevent DoS attacks (253 chars per RFC 1035).
+  static const int _maxFqdnLength = 253;
+
+  /// Maximum length for slug to prevent DoS attacks (63 chars per RFC 1035).
+  static const int _maxSlugLength = 63;
+
+  /// Maximum length for pathname to prevent DoS attacks.
+  static const int _maxPathnameLength = 2048;
```

Comment:
Attach documentation addressing this issue. Write additional unit tests for DoS attacks.